### PR TITLE
chore: improve whisky discoverability on mac

### PR DESCRIPTION
**Pull request by WangEdward**
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/3728#issuecomment-2105946706
Use the same version detection method as [Whisky](https://github.com/Whisky-App/Whisky/blob/57082e3187a4b4481f1c80c85506f939e41bbded/WhiskyKit/Sources/WhiskyKit/WhiskyWine/WhiskyWineInstaller.swift)

Ping @ahmed2m. I do not have Whisky installed. Please download the [CI build](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/actions/runs/9049401139) to see if this works.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by ahmed2m** [comment]
I can't seem to get the CI build working, it just says right after installing
`“Heroic” is damaged and can’t be opened. You should move it to the Trash.`

**Comment by flavioislima** [comment]
> I can't seem to get the CI build working, it just says right after installing `“Heroic” is damaged and can’t be opened. You should move it to the Trash.`

that is because the build file is not signed, you need to run this command from the macOS terminal for it to work:
`xattr -d com.apple.quarantine path_to_Heroic.app`

File: src/backend/utils/compatibility_layers.ts [hunk 1]
```diff
@@ -387,29 +387,32 @@ export async function getWhisky(): Promise<Set<WineInstallation>> {
...
-          name: `Whisky - ${version}`,
-          type: `toolkit`,
-          lib: `${dirname(whiskyWineBin)}/../lib`,
-          lib32: `${dirname(whiskyWineBin)}/../lib`,
-          ...getWineExecs(whiskyWineBin)
-        })
-      }
-    })
-  })
+  const whiskyWinePath = `${userHome}/Library/Application Support/com.isaacmarovitz.Whisky/Libraries`
```
Comment (JoshuaBrest) [inline src/backend/utils/compatibility_layers.ts:390]: use `path.join`

File: src/backend/utils/compatibility_layers.ts [hunk 2]
```diff
@@ -387,29 +387,32 @@ export async function getWhisky(): Promise<Set<WineInstallation>> {
...
+      const info = plistParse(
+        readFileSync(whiskyVersionPlist, 'utf-8')
+      ) as PlistObject
+      const version = info['version']
+      const versionString = `${version['major']}.${version['minor']}.${version['patch']}-${version['build']}`
+      whisky.add({
+        bin: whiskyWineBin,
+        name: `Whisky - ${versionString}`,
+        type: `toolkit`,
+        lib: `${dirname(whiskyWineBin)}/../lib`,
```
Comment (JoshuaBrest) [inline src/backend/utils/compatibility_layers.ts:405]: Use `path.join`

File: src/backend/utils/compatibility_layers.ts [hunk 3]
```diff
@@ -387,29 +387,32 @@ export async function getWhisky(): Promise<Set<WineInstallation>> {
...
+    try {
+      const info = plistParse(
+        readFileSync(whiskyVersionPlist, 'utf-8')
+      ) as PlistObject
+      const version = info['version']
+      const versionString = `${version['major']}.${version['minor']}.${version['patch']}-${version['build']}`
+      whisky.add({
+        bin: whiskyWineBin,
+        name: `Whisky - ${versionString}`,
+        type: `toolkit`,
```
Comment (JoshuaBrest) [inline src/backend/utils/compatibility_layers.ts:404]: Use normal string.

**Review by JoshuaBrest — CHANGES_REQUESTED** [review]

**Comment by ahmed2m** [comment]
It seems that whisky's `CFBundleShortVersionString` in the package contents is not in sync with the Application Support one?

Whisky's latest version is indeed 2.3.2, but what's in the Application Support/Library/WhiskyWineVersion.plist is 2.2.0

Not idea why

---
Current Heroic :
<details>
<summary>Details</summary>

CleanShot 2024-05-20 at 16 43 39@2x


</details>


This MR CI build:

<details>


<summary>Details</summary>

CleanShot 2024-05-20 at 16 44 38@2x


</details>

**Comment by WangEdward** [comment]
> Whisky's latest version is indeed 2.3.2, but what's in the Application Support/Library/WhiskyWineVersion.plist is 2.2.0

The version 2.2.0 is the same as [whisky's wine version](https://github.com/Whisky-App/wine/blob/7.7/WhiskyWineVersion.plist). It does not [update](https://github.com/Whisky-App/wine/commits/7.7/WhiskyWineVersion.plist) wine the same frequency it updates the [launcher](https://github.com/Whisky-App/Whisky/releases)(the .app itself)

File: src/backend/utils/compatibility_layers.ts [hunk 4]
```diff
@@ -387,7 +387,10 @@ export async function getWhisky(): Promise<Set<WineInstallation>> {
     return whisky
   }
 
-  const whiskyWinePath = `${userHome}/Library/Application Support/com.isaacmarovitz.Whisky/Libraries`
+  const whiskyWinePath = join(
+    userHome,
+    'Library/Application Support/com.isaacmarovitz.Whisky/Libraries'
+  )
   const whiskyVersionPlist = `${whiskyWinePath}/WhiskyWineVersion.plist`
   const whiskyWineBin = `${whiskyWinePath}/Wine/bin/wine64`
```
Comment (JoshuaBrest) [inline src/backend/utils/compatibility_layers.ts:395]: Can you update these too?

**Review by flavioislima — APPROVED** [review]
LGTM

**Pull request merged**